roles/vdm/tasks/postgres/postgres-instance.yaml

```diff
@@ -134,6 +173,69 @@
       existing: "{{ vdm_overlays }}"
       add:
         - { transformers: "{{ role }}-crunchy-storage-transformer.yaml", min: "2022.10", vdm: true}
+  - name: postgres instance - set immutable storage transformer values
+    block: 
+    - name: postgres instance - save role storage map values
+      block:
+      - name: postgres instance - save default role map values
+        set_fact:
+          default_postgres_storage_class: "{{ settings.postgres_storage_class|default('pg-storage', true) }}"
+          default_postgres_pvc_access_mode: "{{ settings.postgres_pvc_access_mode|default('ReadWriteOnce', true) }}"
+          default_backrest_storage_class: "{{ settings.backrest_storage_class|default('pg-storage', true) }}"
+          default_backrest_pvc_access_mode: "{{ settings.backrest_pvc_access_mode|default('ReadWriteOnce', true) }}"
+      - name: postgres instance - Create deployment namespace
+        kubernetes.core.k8s:
+          kubeconfig: "{{ KUBECONFIG }}"
+          name: "{{ NAMESPACE }}"
+          api_version: v1
+          kind: Namespace
+          state: present
+      - name: postgres instance - Update ConfigMap
+        kubernetes.core.k8s:
+          kubeconfig: "{{ KUBECONFIG }}"
+          state: present
+          definition:
+            apiVersion: v1
+            kind: ConfigMap
+            metadata:
+              name: dac-crunchy-storage-info
+              namespace: "{{ NAMESPACE }}"
+            data:
+              default-postgres-storage-class: "{{ default_postgres_storage_class | string }}"
+              default-postgres-access-mode: "{{ default_postgres_pvc_access_mode | string }}"
+              default-backrest-storage-class: "{{ default_backrest_storage_class | string }}"
+              default-backrest-access-mode: "{{ default_backrest_pvc_access_mode | string }}"
```

Comment:
This code seems to always update the config map, is that the desired effect here? Thinking we set the values once and then don't set them again ever if we find them.

Reply:
It should not update the config map if the config map already exists, the when clause will prevent the config map from being updated when the dac_crunchy_storage_cm_found flag is set to true.